multistream-select/src/protocol/error.rs

```diff
@@ -44,3 +46,39 @@ impl From<IoError> for MultistreamSelectError {
 		MultistreamSelectError::IoError(err)
 	}
 }
+
+impl error::Error for MultistreamSelectError {
+	#[inline]
+	fn description(&self) -> &str {
+		match *self {
+			MultistreamSelectError::IoError(_) => {
+				"I/O error"
+			},
+			MultistreamSelectError::FailedHandshake => {
+				"the remote doesn't use the same multistream-select protocol as we do"
+			},
+			MultistreamSelectError::UnknownMessage => {
+				"received an unknown message from the remote"
+			},
+			MultistreamSelectError::WrongProtocolName => {
+				"protocol names must always start with `/`, otherwise this error is returned"
+			},
+		}
+	}
+
+	fn cause(&self) -> Option<&error::Error> {
+		match *self {
+			MultistreamSelectError::IoError(ref err) => {
```

Comment:
Why do we repeat `Error`/`MultistreamSelect` twice in this? Can it be `enum Error { Io(...), ... }`? We know we're in the multistream select crate already.